easybuild/easyconfigs/g/GraphicsMagick/GraphicsMagick-1.3.28-GCC-6.4.0-2.28.eb

```diff
@@ -0,0 +1,41 @@
+easyblock = 'ConfigureMake'
+
+name = 'GraphicsMagick'
+version = '1.3.28'
+
+homepage = 'http://www.graphicsmagick.org/'
+description = """GraphicsMagick is the swiss army knife of image processing."""
+
+toolchain = {'name': 'GCC', 'version': '6.4.0-2.28'}
+toolchainopts = {'pic': True}
+
+source_urls = [
+    SOURCEFORGE_SOURCE,
+    'ftp://ftp.graphicsmagick.org/pub/GraphicsMagick/%(version_major_minor)s/',
+]
+sources = [SOURCE_TAR_GZ]
+checksums = ['4b06840d7ce3aee90fde811b42ba2c3594df8cf30bf4620b48c42c8b35d93580']
+
+builddependencies = [('Autotools', '20170619')]
+
+dependencies = [
+    ('X11', '20171023'),
+    ('bzip2', '1.0.6'),
+    ('freetype', '2.8'),
+    ('libpng', '1.6.32'),
+    ('libjpeg-turbo', '1.5.2'),
+    ('LibTIFF', '4.0.9'),
+    ('libxml2', '2.9.4'),
+    ('XZ', '5.2.3'),
+    ('zlib', '1.2.11'),
+    ('Ghostscript', '9.22'),
+]
+
+modextrapaths = {'CPATH': ['include/GraphicsMagick']}
```

Comment:
Is this really needed? Code that uses this is really supposed to do
```
#include <GraphicsMagick/somefile.h>
```

Reply:
I checked the GraphicsMagick codes. It seems at least freetype and X11 are explicitly needed for fundamental pieces in this module. Additionally, this package also has the function to convert/output image format like TIFF, png etc. so that in general they also need these running time library to support these function. Therefore I think it's reasonable to have such a list as dependency.

Reply:
I wasn't talking about the dependencies but about the
```
modextrapaths = {'CPATH': ['include/GraphicsMagick']}
```
line.
By default we always add %(installdir)s/include to CPATH.

Reply:
I see your point. That's reasonable way to define the include path. I changed it into %(installdir)s/include.

Reply:
finally I changed it into modextrapaths = {'CPATH': 'include'} to drop the GraphicsMagick.

Reply:
Still not what i meant.
```
modextrapaths = {'CPATH': 'include'}
```
is what is done by default, which means you don't need it at all.

If you check the resulting module file you should, with the above still in place, see CPATH getting set twice with the same path.

Reply:
OK. Then what do you want me to do? Do you want me to delete this line? 
`modextrapaths = {'CPATH': ['include/GraphicsMagick']}`

Reply:
As I started with, if code that includes files from graphicsmagick does the right thing and uses
```
#include <GraphicsMagick/filename.h>
```
then you do not need the modextrapaths at all. However, if the common way to include files from graphicsmagick is to use
```
#include <filename.h>
```
then you do need
```
modextrapaths = {'CPATH': ['include/GraphicsMagick']}
```
I.e., if we need the modextrapaths or not, depends on how other codes, that use graphicsmagick, does their includes.

Does that make it clear?

Reply:
I understand how the codes look like for the include files, I just do not know what you wants me to do. Next time you can just tell me "delete this line" then I know what you want.

Reply:
delete the include dir, use the default one now.

Reply:
Given the file `include/GraphicsMagick/magick/api.h` - samples on their website [1] uses `#include <magick/api.h>` and the tool `GraphicsMagick-config --cppflags` says `-I/foo/GraphicsMagick`.

As such, the original formulation of `modextrapaths = {'CPATH': ['include/GraphicsMagick']}` would probably be appropriate to mimic the behaviour of existing usage.

[1] http://www.graphicsmagick.org/api/api.html

Reply:
According to the documentation and looking at Octave we need to keep the
```
modextrapaths = {'CPATH': ['include/GraphicsMagick']}
```
line.

Reply:
OK. Add the line back.